llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -278,8 +280,8 @@ class HuggingFaceCheckpointer(Callback):
 
     def __init__(
         self,
-        save_folder: str,
         save_interval: Union[str, int, Time],
+        save_folder: Optional[str] = None,
```

Comment:
its not good to change arg order (technically backwards breaking change if positional args are allowed). would prefer to just make it `Optional` but without a default and keep the same order